src/components/List.js

```diff
@@ -26,12 +28,52 @@ const List = ({ token }) => {
   };
 
   const handleCheckbox = async (event) => {
-    const currentDate = Date.now(); // changed to using Date.now(), time given in milliseconds
+    const currentDate = Date.now(); // time given in milliseconds
     const queryCollection = await db.collection(token).doc(event.target.id);
+    // listItem is our current snapshot of the doc
+    const listItem = docs.filter((doc) => doc.id === queryCollection.id)[0];
+    const previouslyPurchasedDate = listItem.lastPurchased;
+
+    /* latestInterval is the number of full days between the two dates
+     * or the timeFrame if the function returns NaN
+     */
+    const latestInterval =
+      differenceInDays(currentDate, previouslyPurchasedDate) ||
+      listItem.timeFrame;
+
+    // If item has not been purchased set to 1, else increment by 1
+    const numberOfPurchases = (listItem.numberOfPurchases || 0) + 1;
+    const daysUntilNextPurchase = calculateEstimate(
+      listItem.timeFrame,
```

Comment:
This is my only question. My thought for this calculation is that it only uses the `timeFrame` value on the very first purchase, and then subsequently uses the `daysUntilNextPurchase` value. I think the function expects "The last estimated purchase interval", so it makes sense to me to do that because the calculation gets more and more refined as you purchase that item more and more.  What do you all think?

Reply:
@laurenmbeatty 💯  We'll change it so the first purchase uses `timeFrame` and every one after uses `daysUntilNextPurchase` :